lib/dal/src/approval_requirement.rs

```diff
@@ -187,12 +187,9 @@ impl ApprovalRequirementDefinition {
             approvers: content.approvers,
         }
     }
-    pub async fn get_by_id_or_error(
-        ctx: &DalContext,
-        id: ApprovalRequirementDefinitionId,
-    ) -> Result<Self> {
+    pub async fn get_by_id(ctx: &DalContext, id: ApprovalRequirementDefinitionId) -> Result<Self> {
         ctx.workspace_snapshot()?
-            .get_by_id_or_error(ctx, id)
+            .get_by_id(ctx, id)
```

Comment:
While we're renaming this, we should give it a better name in general. This comes from the ApprovalRequirementExt trait but nothing about the function indicates that. It confused me a lot just now staring at it (i assumed it was getting a workspace snapshot by id or erroring?)

Reply:
You are completely right! I assumed the same. This is because `ApprovalRequirementExt` is implemented by WorkspaceSnapshot, letting you treat it as a collection of approval requirements.

The situation is even more confusing, because there are two `ApprovalRequirementExt` traits, each of which is impl'd by exactly one struct. They look sort of similar. I'm not sure what the right turn is here. I'm willing to rename in the short term, though! `get_approval_requirement_by_id()` sounds about right.

Reply:
Fixed.